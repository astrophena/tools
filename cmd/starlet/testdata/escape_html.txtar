-- bot.star --
def send_message(to, text, reply_markup = {}):
    return telegram.call(
        method = "sendMessage",
        args = {
            "chat_id": to,
            "text": text,
            "reply_markup": reply_markup
        }
    )

def handle(update):
    message = '''
    <p>Hello, world!</p>
    <code>%s</code>
    '''
    code = '''
    <h1>PWNED! LALALALAL!</h1>
    '''
    send_message(config.owner_id, message % html.escape(code))
-- update.json --
{
  "update_id": 10000,
  "message": {
    "date": 1441645532,
    "chat": {
      "last_name": "Test Lastname",
      "id": 123456789,
      "type": "private",
      "first_name": "Test Firstname",
      "username": "Testusername"
    },
    "message_id": 1365,
    "from": {
      "last_name": "Test Lastname",
      "id": 123456789,
      "first_name": "Test Firstname",
      "username": "Testusername"
    },
    "forward_from": {
      "last_name": "Forward Lastname",
      "id": 222222,
      "first_name": "Forward Firstname"
    },
    "forward_date": 1441645550,
    "text": "/start"
  }
}
