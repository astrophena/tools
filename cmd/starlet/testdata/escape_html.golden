[
  {
    "method": "sendMessage",
    "args": {
      "chat_id": 123456789,
      "reply_markup": {},
      "text": "\n    \u003cp\u003eHello, world!\u003c/p\u003e\n    \u003ccode\u003e\n    \u0026lt;h1\u0026gt;PWNED! LALALALAL!\u0026lt;/h1\u0026gt;\n    \u003c/code\u003e\n    "
    }
  }
]